dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java

```diff
@@ -0,0 +1,140 @@
+        builder.lastModifiedBy(user.getUserId());
+
+        final Experiment experimentToSave = builder.build();
+
+        return factory.save(experimentToSave);
+    }
+
+    @Override
+    public Optional<Experiment> find(final String id, final User user)
+            throws DotDataException, DotSecurityException {
+        if(!UtilMethods.isSet(id)) {
+            return Optional.empty();
+        }
+
+        final Optional<Experiment> experiment =  factory.find(id);
+
+        if(experiment.isPresent()) {
+            final Contentlet pageAsContent = contentletAPI
+                    .findContentletByIdentifierAnyLanguage(experiment.get().pageId(), false);
+
+            if (!permissionAPI.doesUserHavePermission(pageAsContent, PermissionLevel.EDIT.getType(),
+                    user)) {
+                Logger.error(this, "You don't have permission to get the Experiment.");
+                throw new DotSecurityException("You don't have permission to get the Experiment.");
+            }
+        }
+
+        return experiment;
+    }
+
+    @Override
+    public Experiment archive(final String id, final User user)
+            throws DotDataException, DotSecurityException {
+        DotPreconditions.checkArgument(UtilMethods.isSet(id), "id must be provided.");
+
+        final Optional<Experiment> persistedExperiment =  factory.find(id);
+
+        if(persistedExperiment.isPresent()) {
+            final Contentlet pageAsContent = contentletAPI
+                    .findContentletByIdentifierAnyLanguage(persistedExperiment.get().pageId(), false);
+
+            if (!permissionAPI.doesUserHavePermission(pageAsContent, PermissionLevel.EDIT.getType(),
+                    user)) {
+                Logger.error(this, "You don't have permission to get the Experiment.");
+                throw new DotSecurityException("You don't have permission to get the Experiment.");
+            }
+
+            if(persistedExperiment.get().status()!= Status.ENDED) {
+                throw new DotStateException("Only ended experiments can be archived");
+            }
+
+            final Experiment archived = persistedExperiment.get().withArchived(true);
+            return factory.save(archived);
+        } else {
+            throw new NotFoundInDbException("Experiment with provided id not found");
+        }
+
+    }
+
+    @Override
+    public void delete(final String id, final User user)
+            throws DotDataException, DotSecurityException {
+        DotPreconditions.checkArgument(UtilMethods.isSet(id), "id must be provided.");
+
+        final Optional<Experiment> persistedExperiment =  factory.find(id);
+
+        if(persistedExperiment.isPresent()) {
+            final Contentlet pageAsContent = contentletAPI
+                    .findContentletByIdentifierAnyLanguage(persistedExperiment.get().pageId(), false);
+
+            if (!permissionAPI.doesUserHavePermission(pageAsContent, PermissionLevel.EDIT.getType(),
```

Comment:
use proxy and more log

Reply:
also if the user is admin, should not do any check, right?

Reply:
This happens inside the `doesUserHavePermission`

Reply:
Done